inc/class-purge-cache.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * WP Irving Cache.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving;
+
+/**
+ * Class to purge cache.
+ */
+class Purge_Cache {
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		// Register admin page.
+		add_action( 'admin_menu', [ $this, 'register_admin' ] );
+
+		// Purging actions.
+		add_action( 'wp_insert_post', [ $this, 'on_update_post' ], 10, 2 );
+		add_action( 'transition_post_status', [ $this, 'on_post_status_transition' ], 10, 3 );
```

Comment:
- Is there a situation where `wp_insert_post` would fire but `transition_post_status` would not fire? Only the `transition_post_status` hook might be necessary.
- Also, does the cache need to be purged when non-viewable post types change status?